control-plane-pki-operator/certificaterevocationcontroller/certificaterevocationcontroller.go

```diff
@@ -535,6 +543,84 @@ func (c *CertificateRevocationController) generateNewSignerCertificate(ctx conte
 	return false, nil, false, nil
 }
 
+// testCertificateAgainstAllKASPods tests a certificate against all kube-apiserver pods
+// and returns true if the test passes for all pods, false otherwise.
+// The testFunc should return true if the test passes, false if it should requeue.
+func (c *CertificateRevocationController) testCertificateAgainstAllKASPods(
+	ctx context.Context,
+	namespace string,
+	certPEM, keyPEM []byte,
+	testFunc func(client kubernetes.Interface) (bool, error),
+) (bool, error) {
+	if c.skipKASConnections {
+		return true, nil
+	}
+
+	// Get the base kubeconfig for the guest cluster service network
+	kubeconfig := hcpmanifests.KASServiceKubeconfigSecret(namespace)
+	kubeconfigSecret, err := c.getSecret(kubeconfig.Namespace, kubeconfig.Name)
+	if err != nil {
+		return false, fmt.Errorf("couldn't fetch guest cluster service network kubeconfig: %w", err)
+	}
+	adminClientCfg, err := clientcmd.NewClientConfigFromBytes(kubeconfigSecret.Data["kubeconfig"])
+	if err != nil {
+		return false, fmt.Errorf("couldn't load guest cluster service network kubeconfig: %w", err)
+	}
+	baseCfg, err := adminClientCfg.ClientConfig()
+	if err != nil {
+		return false, fmt.Errorf("couldn't load guest cluster service network kubeconfig: %w", err)
+	}
+
+	// List all kube-apiserver pods
+	pods, err := c.listPods(namespace, "app=kube-apiserver")
+	if err != nil {
+		return false, fmt.Errorf("couldn't list kube-apiserver pods: %w", err)
+	}
+	if len(pods) == 0 {
+		// No pods found, requeue
+		return false, nil
+	}
+
+	// Test the certificate against each kube-apiserver pod
+	for _, pod := range pods {
+		if pod.Status.Phase != corev1.PodRunning {
+			// Pod is not running, requeue to wait for it
+			return false, nil
+		}
+		if pod.Status.PodIP == "" {
+			// Pod doesn't have an IP yet, requeue
+			return false, nil
```

Comment:
We requeue

Reply:
Logs might be nice, but timeout and requeue interval is handled by the controller queue itself

Reply:
Thanks @stevekuznetsov